fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/aggregate/AggregateCursor.java

```diff
@@ -42,54 +42,56 @@
  * @param <M> the record type that holds the actual data
  */
 @API(API.Status.EXPERIMENTAL)
-public class AggregateCursor<M extends Message> implements RecordCursor<QueryResult> {
+public class AggregateCursor<M extends Message> implements RecordCursor<Object> {
```

Comment:
This seems odd. Even though multi-column records are now a single value, I would not expect this to have to change.

Looked at another way, this cursor is doing two things: extracting a value from the input cursor and aggregating. I would expect it to either be `RecordCursor<QueryResult>` -> `RecordCursor<QueryResult>` if operating on result sets or `RecordCursor<Object>` -> `RecordCursor<Object>` if operating on aggregatable values, requiring the calling operator to supply the two `map`s. In the latter case, better might be `RecordCursor<M1 extends Message>` -> `RecordCursor<M2 extends Message>` with `StreamingGroup<M1, M2>`.

Reply:
I changed it back to `RecordCursor<QueryResult>` so it can directly be used by the plan operator `RecordQueryStreamingAggregationPlan`. So in effect `StreamGrouping` aggregates `Value`s and therefore `Object` which is in line with what other `Value`-based operators like filter and also map, `AggregateCursor` does the extraction/repackaging from/to `QueryResult`.